apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -99,6 +99,55 @@ export const pageRouter = router({
         .execute()
     }),
 
+  getCategories: protectedProcedure
+    .input(basePageSchema)
+    .query(async ({ ctx, input: { pageId, siteId } }) => {
+      await validateUserPermissionsForResource({
+        userId: ctx.user.id,
+        siteId,
+        action: "read",
+      })
+
+      const { parentId } = await db
+        .selectFrom("Resource")
+        .where("id", "=", String(pageId))
+        .where("type", "in", [
+          ResourceType.CollectionPage,
+          ResourceType.CollectionLink,
+        ])
+        .select("parentId")
+        .executeTakeFirstOrThrow()
+
+      const blobs = await db
+        .selectFrom("Resource as r")
+        .leftJoin("Blob as b", "r.draftBlobId", "b.id")
+        .leftJoin("Version as v", "r.publishedVersionId", "v.id")
+        .leftJoin("Blob as vb", "v.blobId", "vb.id")
+        .where("r.type", "in", [
+          ResourceType.CollectionPage,
+          ResourceType.CollectionLink,
+        ])
+        .select((eb) => {
+          return eb.fn
+            .coalesce(
+              sql<string>`b.content->'page'->>'category'`,
```

Comment:
for future: might wanna add indexes on `content->'page'->>'category'`

Reply:
i can do this in a follow-up, no worries

Reply:
addon - potentially wanna create a autogenerated column on this and index this autogen column -> easier to reference and search for it in the codebase than calling `content->'page'->>'category'` + easier to identify it when its unused thus safer to remove